components/board/genericlinux/data.go

```diff
@@ -144,53 +144,40 @@ func getPwmChipDefs(pinDefs []PinDefinition) (map[string]pwmChipData, error) {
 
 	// Now, look for all chips whose names we found.
 	pwmChipsInfo := map[string]pwmChipData{}
+	var found bool
+	sysPrefix := "/sys/class/pwm"
+	files, err := os.ReadDir(sysPrefix)
+
 	for chipName := range pwmChipNames {
-		found := false
-
-		// The boards we support might store their PWM devices in several different locations
-		// within /sys/devices. If it turns out that every board stores its PWM chips in a
-		// different location, we should rethink this. Maybe a breadth-first search over all of
-		// /sys/devices/? but for now, this is good enough.
-		directoriesToSearch := []string{
-			// Example boards which use each location:
-			// Jetson Orin AGX       BeagleBone AI64                     Intel UP 4000
-			"/sys/devices/platform", "/sys/devices/platform/bus@100000", "/sys/devices/pci0000:00",
+		found = false
+		if err != nil {
+			return nil, err
 		}
-		for _, baseDir := range directoriesToSearch {
-			// For exactly one baseDir, there should be a directory at <baseDir>/<chipName>/pwm/,
-			// which contains a single sub-directory whose name is mirrored in /sys/class/pwm.
-			// That's the one we want to use.
-			chipDir := fmt.Sprintf("%s/%s/pwm", baseDir, chipName)
-			files, err := os.ReadDir(chipDir)
+
+		for _, file := range files {
+			if !strings.HasPrefix(file.Name(), "pwmchip") {
+				continue
+			}
+
+			// look at symlinks to find the correct chip
+			symlink, err := os.Readlink(filepath.Join(sysPrefix, file.Name()))
 			if err != nil {
-				continue // This was the wrong directory; try the next baseDir.
+				return nil, err
```

Comment:
Could we ever have a non-symlink file in here? I'm unsure, but that would give back an "invalid argument" error. In that case, I'm inclined to merely log the error, and then continue on to the next file, rather than giving up completely.